### PR TITLE
Merge Hadoop projects into a single project

### LINKED ISSUE #1438: Merge Hadoop projects into a single project
We only support Hadoop which is supported by Tinkerpop, so we can merge all Hadoop project together.

_That's one small step for a man, one giant leap for JanusGraph_ 
**Yeah cleanup :tada:**

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #1438

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

I compared the release bundled zip with old bundled zip and it seems contain all data as before.

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-dist/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ SNAPSHOT or a release tag.
 
 This command writes one archive:
 
-* janusgraph-dist/janusgraph-dist-hadoop-2/target/janusgraph-$VERSION-hadoop2.zip
+* janusgraph-dist/target/janusgraph-$VERSION.zip
```
Comment (chupman) [inline janusgraph-dist/README.md]: While this is a much better name and path for the distribution, this change will totally throw me off for a good chunk of time.

**Comment by farodin91** [comment]
@porunov @chupman Do you like to review this PR?

File: janusgraph-hadoop/pom.xml [hunk 2]
```diff
@@ -6,13 +6,12 @@
         <version>0.5.0-SNAPSHOT</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
-    <artifactId>janusgraph-hadoop-parent</artifactId>
-    <name>JanusGraph-Hadoop: Parent Module</name>
-    <packaging>pom</packaging>
-    <url>https://janusgraph.org</url>
+    <artifactId>janusgraph-hadoop</artifactId>
+    <name>JanusGraph-Hadoop: Module</name>
+    <url>http://janusgraph.org</url>
```
Comment (porunov) [inline janusgraph-hadoop/pom.xml]: `http` > `https`

File: janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/compat/HadoopCompatLoader.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+public class HadoopCompatLoader {
+
+
+    public static final HadoopCompat DEFAULT_COMPAT = getCompat();
+
+    // TODO add a string argument that allows specifying a class instead of relying heuristics around VersionInfo.getVersion()
+    // TODO add threadsafe caching that is aware of the string argument and instantiates a compat for each argument at most once (assuming the instantiation succeeds)
+    public static HadoopCompat getCompat() {
+        return new Hadoop2Compat();
```
Comment (porunov) [inline janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/compat/HadoopCompatLoader.java]: In the previous `HadoopCompatLoader` it was returning either `Hadoop1Compat` or `Hadoop2Compat` depending on `VersionInfo.getVersion()` and now it is always `Hadoop2Compat`. Is it OK?

Comment (farodin91) [reply janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/compat/HadoopCompatLoader.java]: I will remove the compat completely. We only support hadoop2 for long time. I would prefer to support only tinkerpop supported version of Hadoop.

Comment (porunov) [reply janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/compat/HadoopCompatLoader.java]: I am OK dropping support for Hadoop 1 (if it isn't dropped yet) but is it a breaking change? If so, should this be reflected in `upgrade.adoc`?

Comment (farodin91) [reply janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/compat/HadoopCompatLoader.java]: I will check the upgrade doc, but this pr will be followed by another PR to invert order of Hadoop and backends

File: janusgraph-hadoop/pom.xml (hunk 2 above)
Comment (porunov) [reply janusgraph-hadoop/pom.xml]: Still not changed

**Comment by farodin91** [comment]
@porunov I did rebase on the master again. Do you like to check it out again?

File: janusgraph-hadoop/pom.xml (hunk 2 above)
Comment (porunov) [inline janusgraph-hadoop/pom.xml]: Should be `<url>https://janusgraph.org</url>`

**Pull request merged**